test/suites/test_datetime_arithmetic.py

```diff
@@ -0,0 +1,317 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function
+
+import sys
+import unittest
+import msgpack
+import warnings
+import tarantool
+import pandas
+import pytz
+
+from tarantool.msgpack_ext.packer import default as packer_default
+from tarantool.msgpack_ext.unpacker import ext_hook as unpacker_ext_hook
+
+from .lib.tarantool_server import TarantoolServer
+from .lib.skip import skip_or_run_datetime_test
+from tarantool.error import MsgpackError
+
+class TestSuite_DatetimeArithmetic(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        print(' DATETIME TYPE ARITHMETIC '.center(70, '='), file=sys.stderr)
+        print('-' * 70, file=sys.stderr)
+        self.srv = TarantoolServer()
+        self.srv.script = 'test/suites/box.lua'
+        self.srv.start()
+
+        self.adm = self.srv.admin
+        self.adm(r"""
+            _, datetime = pcall(require, 'datetime')
+
+            box.schema.space.create('test')
+            box.space['test']:create_index('primary', {
+                type = 'tree',
+                parts = {1, 'string'},
+                unique = true})
+
+            box.schema.user.create('test', {password = 'test', if_not_exists = true})
+            box.schema.user.grant('test', 'read,write,execute', 'universe')
+
+            local function add(arg1, arg2)
+                return arg1 + arg2
+            end
+            rawset(_G, 'add', add)
+
+            local function sub(arg1, arg2)
+                return arg1 - arg2
+            end
+            rawset(_G, 'sub', sub)
+        """)
+
+        self.con = tarantool.Connection(self.srv.host, self.srv.args['primary'],
+                                        user='test', password='test')
+
+    def setUp(self):
+        # prevent a remote tarantool from clean our session
+        if self.srv.is_started():
+            self.srv.touch_lock()
+
+        self.adm("box.space['test']:truncate()")
+
+
+    interval_cases = {
```

Comment:
It looks like tests for the Interval class, not for the Datetime.

Reply:
By "Datetime arithmetic" and I implied both datetime and interval interactions. I think it is alright since in Tarantool the classes are built with `datetime.new` and `datetime.interval` new, but if you have some ideas about what name is better suited to describe both of them -- I would be glad to hear it.

Reply:
I think it will be more obvious to leave two files: `test_datetime.py` for the Datetime class API and `test_interval.py` for the Interval class API. As example I have three files:

1. test_datetime.py
2. test_interval.py
3. test_datetime_arithmetic.py

Where should I look test for `+` operation of the Interval class?

Reply:
Well, now all `+` tests is in `test_datetime_arithmetic.py`.

With your approach, it would be `Interval + Interval` in `test_interval.py` and `Datetime + Interval` in... `test_interval.py`? `test_datetime.py`? Separate file `test_datetime_arithmetic.py`? To be honest, I don't think there could be non-confusing approaches.

Reply:
test_datetime.py: `+` is the method of the Datetime class.

Reply:
Well, the solution was more obvious than I expected.

Moved all interval methods tests to `test_interval.py` and datetime methods tests to `test_datetime.py`